File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        
        /*for(int i = 0; i<nums.length; i++){
            if(target == nums[i]){
                return i;
            }
        }

        return -1;
        */

        int l = 0;
        int r = nums.length-1;
        int mid;

        while(l<=r){
            mid = l+(r-l)/2;

            if(nums[mid] > target){
                r=mid-1;
            }

            else if(nums[mid] < target){
                l=mid+1;
            }

            else{
                return mid;
            }

        }

        return -1;
    }
}

File: 0792-binary-search/0792-binary-search.cpp
class Solution {
public:
    
    int binarySearch(vector<int>& nums, int l, int r, int target){

        if(l > r) return -1;

        int mid = l + (r-l)/2;

        if(nums[mid] < target){
            return binarySearch(nums, mid+1, r, target);
        }
        else if(nums[mid] > target){
            return binarySearch(nums, l, mid-1, target);
        }
        else{
            return mid;
        }
    }


    int search(vector<int>& nums, int target) {
        
        int l = 0;
        int r = nums.size()-1;

        return binarySearch(nums,l,r,target);
    }
};